internal/build/phases.go

```diff
@@ -30,13 +39,15 @@ func (l *Lifecycle) Detect(ctx context.Context, networkMode string, volumes []st
 	if err != nil {
 		return err
 	}
+
+	detect := phaseFactory.New(configProvider)
 	defer detect.Cleanup()
 	return detect.Run(ctx)
 }
 
-func (l *Lifecycle) Restore(ctx context.Context, cacheName string) error {
-	restore, err := l.NewPhase(
-		"restorer",
+func (l *Lifecycle) Restore(ctx context.Context, cacheName string, phaseFactory PhaseFactory) error {
+	configProvider, err := NewPhaseConfigProvider(
+		"restorer", l,
```

Comment:
We should probably have `l` on a separate line, even though that might seem like a waste of space, I think visually when the rest of the args are on separate lines it's easy to notice there are two on this line.